numpy/_core/numerictypes.py

```diff
@@ -359,6 +359,72 @@ def issubsctype(arg1, arg2):
     return issubclass(obj2sctype(arg1), obj2sctype(arg2))
 
 
+@set_module('numpy')
+def isdtype(input_dtype, kind):
+    """
+    Returns a boolean indicating whether a provided dtype is of a specified
+    data type ``kind``.
+
+    Parameters
+    ----------
+    input_dtype : dtype
+        The input dtype.
+    kind : dtype or str or tuple of dtypes/strs.
+        dtype or dtype kind. Allowed dtype kinds are:
+        * ``bool`` : boolean kind
+        * ``signed integer`` : signed integer data types
+        * ``unsigned integer`` : unsigned integer data types
+        * ``integral`` : integer data types
+        * ``real floating`` : real-valued floating-point data types
+        * ``complex floating`` : complex floating-point data types
+        * ``numeric`` : numeric data types
+
+    Returns
+    -------
+    out : bool
+
+    See Also
+    --------
+    issubdtype
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> np.isdtype(np.int64, (np.uint64, "signed integer"))
+    True
+    >>> np.isdtype(np.float32, np.float64)
+    False
+
+    """
+    input_dtype = dtype(input_dtype).type
+    input_kinds = kind if isinstance(kind, tuple) else (kind,)
+
+    processed_kinds = set()
+
+    for kind in input_kinds:
+        if kind == "bool":
+            processed_kinds.add(allTypes["bool_"])
+        elif kind == "signed integer":
+            processed_kinds.update(sctypes["int"])
+        elif kind == "unsigned integer":
+            processed_kinds.update(sctypes["uint"])
+        elif kind == "integral":
+            processed_kinds.update(sctypes["int"] + sctypes["uint"])
+        elif kind == "real floating":
+            processed_kinds.update(sctypes["float"])
+        elif kind == "complex floating":
+            processed_kinds.update(sctypes["complex"])
+        elif kind == "numeric":
+            processed_kinds.update(
+                sctypes["int"] + sctypes["uint"] +
+                sctypes["float"] + sctypes["complex"]
+            )
+        else:
+            processed_kinds.add(dtype(kind).type)
+
+    return any([input_dtype == k for k in processed_kinds])
```

Comment:
Just `input_dtype in procecessed_kinds` should work.

Reply:
Thanks! I updated it.